frontend/benefit/handler/src/components/applicationList/useApplicationListData.ts

```diff
@@ -58,25 +59,47 @@ const useApplicationListData = (
         applicationNum,
         // refactor when we have handler data
         handlerName:
-          getFullName(
+          `${getFullName(
             calculation?.handler_details?.first_name,
-            calculation?.handler_details?.last_name
-          ) || '-',
+            calculation?.handler_details?.last_name.at(0)
+          )}.` || '-',
         unreadMessagesCount: unread_messages_count ?? 0,
         batch: batch ?? null,
         applicationOrigin,
         talpaStatus: talpa_status,
         ahjoCaseId: ahjo_case_id,
         handledByAhjoAutomation: handled_by_ahjo_automation,
+        handledAt: convertToUIDateFormat(handledAt) || '-',
       };
     })
     .filter(
       (application) =>
-        isNewAhjoMode ||
-        (![
+        [APPLICATION_STATUSES.DRAFT, APPLICATION_STATUSES.RECEIVED].includes(
+          application.status
+        ) ||
+        ([
+          APPLICATION_STATUSES.HANDLING,
+          APPLICATION_STATUSES.INFO_REQUIRED,
+        ].includes(application.status) &&
+          isNewAhjoMode &&
+          application.handledByAhjoAutomation) ||
+        ([
+          APPLICATION_STATUSES.ACCEPTED,
+          APPLICATION_STATUSES.REJECTED,
+        ].includes(application.status) &&
+          isNewAhjoMode &&
+          application.handledByAhjoAutomation) ||
+        ([
```

Comment:
I think these two can be combined together at least to make this huge effective one-liner a bit less wieldy. Could consider just abandoning the idea of combining this all into one statement and using a function with actual if-else structure, too.